src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -92,6 +92,13 @@ public function load(array $configs, ContainerBuilder $container)
             $dir = dirname($reflection->getFileName()).'/Resources/views';
             if (is_dir($dir)) {
                 $this->addTwigPath($twigFilesystemLoaderDefinition, $dir, $bundle);
+
+                $classMock = new $class();
+                $parent = $classMock->getParent();
+
+                if ($parent) {
```

Comment:
Also if you have A>B>C, asking the parents for A should resolve the parents for B also. So if you ask for that later it's there.